src/acc/libsmm_acc/libcusmm/predict_helpers.py

```diff
@@ -0,0 +1,223 @@
+# -*- coding: utf-8 -*-
+####################################################################################################
+# Copyright (C) by the DBCSR developers group - All rights reserved                                #
+# This file is part of the DBCSR library.                                                          #
+#                                                                                                  #
+# For information on the license, see the LICENSE file.                                            #
+# For further information please visit https://dbcsr.cp2k.org                                      #
+# SPDX-License-Identifier: GPL-2.0+                                                                #
+####################################################################################################
+
+import sys
+import os
+import pickle
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from kernels.cusmm_predict import to_string
+
+
+# ===============================================================================
+# I/O helpers
+def safe_pickle(data, file):
+    """
+    Pickle big files safely by processing them in chunks
+    :param data: data to be pickled
+    :param file: file to pickle it into
+    """
+    max_bytes = 2**31 - 1  # Maximum number of bytes to pickle in one chunk
+    pickle_out = pickle.dumps(data)
+    n_bytes = sys.getsizeof(pickle_out)
```

Comment:
are you sure that `len(pickle_out)` doesn't work? You have a `bytes` object and you're writing the contained `bytes`, while `sys.getsizeof` gives the size of the python object including the metadata. Compare:

```python
import pickle
import sys

d = {'foo': 123}
print(len(pickle.dumps(d)))
print(sys.getsizeof(pickle.dumps(d)))
```

Reply:
Good catch! 
I believe my error doesn't necessarily introduce buggy behaviour though, since `sys.getsizeof` returns a value larger than `len` and `f.read` in `safe_pickle_load` only reads until EOF.
Thanks for that!